cypress/support/step_definitions/fireboltCalls.js

```diff
@@ -411,6 +413,94 @@ Given(/I clear '(.+)' listeners$/, async (key) => {
+ * @example
+ * And Fetch response for 'pinChallenge onRequestChallenge' event from '1st party app'
+ * And Fetch response for 'profile approvePurchase' method from '3rd party app'
+ */
+
+Given(
+  /Fetch response for '(.+)' (method|event) from (3rd party app|1st party app)$/,
+  async (key, methodOrEvent, app) => {
+    cy.fireboltDataParser(key).then((parsedDataArr) => {
+      parsedDataArr.forEach((parsedData) => {
+        const method = parsedData.method;
+        const param = parsedData.params;
+        const context = parsedData.context;
+        const action = parsedData.action;
+        const expected = parsedData.expected;
+
+        const appId = UTILS.getEnvVariable(CONSTANTS.THIRD_PARTY_APP_ID);
+
+        let params;
+        if (app == CONSTANTS.FIRST_PARTY_APP) {
+          const extractedEvent = UTILS.getEnvVariable(CONSTANTS.GLOBAL_EVENT_OBJECT_LIST).filter(
+            (element) => element.eventName == method
+          );
+          eventName = extractedEvent[extractedEvent.length - 1].eventObjectId;
+          const requestMap = {
+            method: CONSTANTS.REQUEST_OVERRIDE_CALLS.FETCH_EVENT_RESPONSE,
+            params: eventName,
+          };
+          cy.log(
+            'Call from 1st party App, method: ' + method + ' params: ' + JSON.stringify(params)
+          );
+          // Sending message to first party app.
+          cy.sendMessagetoPlatforms(requestMap).then((response) => {
+            cy.log('Response from Firebolt platform: ' + JSON.stringify(response));
+            if (response === CONSTANTS.RESPONSE_NOT_FOUND) {
+              cy.log(CONSTANTS.NO_MATCHED_RESPONSE).then(() => {
+                assert(false, CONSTANTS.NO_MATCHED_RESPONSE);
+              });
+            }
+            // saving the correlationId of rpc-only methods
+            if (Cypress.env(CONSTANTS.IS_RPC_ONLY)) {
+              cy.log(`correlationId - ${response.result.correlationId}`);
+              Cypress.env(CONSTANTS.CORRELATIONID, response.result.correlationId);
+            }
+          });
+        } else if (app == CONSTANTS.THIRD_PARTY_APP) {
+          params = { method: method };
+
+          // Creating intent message using above details to send it to 3rd party app.
+          const parsedIntent = UTILS.createIntentMessage(CONSTANTS.TASK.GETMETHODRESPONSE, params);
+          // Fetching method response from third party app
+          const requestTopic = UTILS.getTopic(appId);
+          const responseTopic = UTILS.getTopic(appId, CONSTANTS.SUBSCRIBE);
+
+          // Sending message to 3rd party app.
+          cy.sendMessagetoApp(requestTopic, responseTopic, parsedIntent).then((response) => {
+            if (response === CONSTANTS.RESPONSE_NOT_FOUND) {
+              cy.log(CONSTANTS.NO_MATCHED_RESPONSE).then(() => {
+                assert(false, CONSTANTS.NO_MATCHED_RESPONSE);
+              });
+            }
+            cy.log(`Updated response of ${method}: ${JSON.stringify(response)}`);
```

Comment:
Log can be changed to 
**_${method} response from comcast.test.firecert_**

Reply:
Updated the log as :
            cy.log(`${method} response from ${appId}: ${JSON.stringify(response)}`);